serializer-legacy/src/main/java/net/kyori/text/serializer/legacy/LegacyComponentSerializerImpl.java

```diff
@@ -25,19 +25,19 @@
 
 import net.kyori.text.Component;
 import net.kyori.text.TextComponent;
+import net.kyori.text.event.ClickEvent;
 import net.kyori.text.format.TextColor;
 import net.kyori.text.format.TextDecoration;
 import net.kyori.text.format.TextFormat;
 import org.checkerframework.checker.nullness.qual.NonNull;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 class LegacyComponentSerializerImpl implements LegacyComponentSerializer {
+  private static final Pattern URL_PATTERN = Pattern.compile("(?:(https?)://)?([-\\w_.]{2,}\\.[a-z]{2,4})(/\\S*)?");
```

Comment:
One char subdomains are pretty common (especially for image servers, e.g. `i.imgur.com`. I believe the current regex only allows for 2+ char subdomains.

Same thing for TLDs. gTLDs longer than 4 chars are becoming pretty common (.shopping, .tech, etc.).

Maybe not supporting punnycode domains is a good idea (I don't even know how the MC font renders these chars).

This PR looks good to me except the regex (which can be changed). URL parsing is a hard problem 😛.

Reply:
Extreme minor nit: mind changing the a-z to \w (the same way the subdomains are matched).

This regex also doesn't match on numeric domains (I believe the spec allows the full ASCII charset except control chars maybe, don't quote me on it)

Reply:
[This handles really short urls fine](https://github.com/mikroskeem/Adminchatter/blob/master/common/src/main/kotlin/eu/mikroskeem/adminchatter/common/utils/Chat.kt#L61)

Reply:
Aaaand another pattern to throw into the mix ;)

https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/browse/src/main/java/org/bukkit/craftbukkit/util/CraftChatMessage.java#21